### PR TITLE
move benchmarks to separate module 'stresstest'

### PR DESCRIPTION
**Pull request by mfelsche**
and created small framework for running tests concurrently

### CODE REVIEW

File: stresstest/build.gradle [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    compile project(':sql')
+    testCompile files(project(':sql').sourceSets.test.output.classesDir)
+    testCompile 'com.carrotsearch:junit-benchmarks:0.7.2'
+    testCompile 'com.h2database:h2:1.3.173'
+    testCompile group: 'junit', name: 'junit', version: '4.12'
+}
+
+test {
+    enabled = false
+    jacoco {
```
Comment (mfussenegger) [inline stresstest/build.gradle]: do we need/want coverage data from stresstest?

File: stresstest/src/test/java/io/crate/benchmark/SelectBenchmark.java [hunk 2]
```diff
@@ -179,7 +176,7 @@ public void testGetSingleResultApi() {
         for (int i=0; i<NUM_REQUESTS_PER_TEST; i++) {
             GetRequest request = getApiGetRequest(false);
             GetResponse response = getClient(false).execute(GetAction.INSTANCE, request).actionGet();
-            assertTrue(String.format(Locale.ENGLISH, "Queried row '%s' does not exist (API). Round: %d", request.id(), apiGetRound), response.isExists());
+            Assert.assertTrue(String.format(Locale.ENGLISH, "Queried row '%s' does not exist (API). Round: %d", request.id(), apiGetRound), response.isExists());
```
Comment (mfussenegger) [inline stresstest/src/test/java/io/crate/benchmark/SelectBenchmark.java]: why remove the static imports?

File: stresstest/src/test/java/io/crate/concurrent/ThreadedExampleTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    @After
+    public void cleanUp() {
+        System.out.println("@After: " + Thread.currentThread().getName());
+    }
+
+    @Threaded(count = 2)
+    @Test
+    public void testThreaded() throws Exception {
+        String result = TestingHelpers.printedTable(execute("select * from sys.cluster").rows());
+        System.out.println(result);
```
Comment (mfussenegger) [inline stresstest/src/test/java/io/crate/concurrent/ThreadedExampleTest.java]: no assertion or something? does this test even make sense in this form?

Comment (mfelsche) [inline stresstest/src/test/java/io/crate/concurrent/ThreadedExampleTest.java]: as the name suggests its only an example

File: stresstest/src/test/java/io/crate/concurrent/ThreadedStatement.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            final CountDownLatch startLatch = new CountDownLatch(count);
+            ExecutorService service = Executors.newFixedThreadPool(count, EsExecutors.daemonThreadFactory(ImmutableSettings.EMPTY, "Threaded"));
+            ExecutorCompletionService<Void> completionService = new ExecutorCompletionService<>(service);
+            try {
+                for (int i = 0; i < count; i++) {
+                    completionService.submit(new Runnable() {
+                        @Override
+                        public void run() {
+                            try {
+                                startLatch.await(); // all threads wait for each other
```
Comment (mfussenegger) [inline stresstest/src/test/java/io/crate/concurrent/ThreadedStatement.java]: why do you want to start all threads at the same time?

Comment (mfelsche) [inline stresstest/src/test/java/io/crate/concurrent/ThreadedStatement.java]: in order to increase the possibility that things actually happen concurrently

File: stresstest/src/test/java/io/crate/concurrent/ThreadedExampleTest.java (hunk 3 above)
Comment (mfussenegger) [inline stresstest/src/test/java/io/crate/concurrent/ThreadedExampleTest.java]: don't think that we need that..

**Comment by mfussenegger** [comment]
ack

**Pull request merged**